### ISSUE TITLE
kvm full system workloads fail to run on x86

### ISSUE THREAD

**Issue by Harshil2107** [gem5/gem5#1126]
**Describe the bug**
This bug was introduced in #1065 
KVM full system workload fail with the following error:
```
build/X86/gem5.opt configs/example/gem5_library/test_script.py 
gem5 Simulator System.  https://www.gem5.org/
gem5 is copyrighted software; use the --copyright option for details.
 
gem5 version DEVELOP-FOR-24.0
gem5 compiled May 10 2024 11:06:09
gem5 started May 10 2024 11:10:58
gem5 executing on COE-CS-sterling, pid 65871
command line: build/X86/gem5.opt configs/example/gem5_library/test_script.py
 
info: Using default config
Resource 'x86-ubuntu-22.04-img' was not found locally. Downloading to '/home/harshilp/.cache/gem5/x86-ubuntu-22.04-img.gz'...
Finished downloading resource 'x86-ubuntu-22.04-img'.
Decompressing resource 'x86-ubuntu-22.04-img' ('/home/harshilp/.cache/gem5/x86-ubuntu-22.04-img.gz')...
Finished decompressing resource 'x86-ubuntu-22.04-img'.
warn: <orphan X86Board>.kvm_vm already has parent not resetting parent.
        Note: kvm_vm is not a parameter of X86Board
warn: (Previously declared as <orphan X86Board>.processor.kvm_vm
Global frequency set at 1000000000000 ticks per second
warn: No dot file generated. Please install pydot to generate the dot file and pdf.
src/mem/dram_interface.cc:690: warn: DRAM device capacity (8192 Mbytes) does not match the address range assigned (4096 Mbytes)
src/sim/kernel_workload.cc:46: info: kernel located at: /home/harshilp/.cache/gem5/x86-linux-kernel-5.4.0-105-generic
src/base/statistics.hh:279: warn: One of the stats is a legacy stat. Legacy stat is a stat that does not belong to any statistics::Group. Legacy stat is deprecated.
      0: board.pc.south_bridge.cmos.rtc: Real-time clock set to Sun Jan  1 00:00:00 2012
board.pc.com_1.device: Listening for connections on port 3456
src/base/statistics.hh:279: warn: One of the stats is a legacy stat. Legacy stat is a stat that does not belong to any statistics::Group. Legacy stat is deprecated.
src/dev/intel_8254_timer.cc:128: warn: Reading current count from inactive timer.
board.remote_gdb: Listening for connections on port 7000
src/cpu/kvm/base.cc:170: info: KVM: Coalesced MMIO disabled by config.
src/cpu/kvm/base.cc:170: info: KVM: Coalesced MMIO disabled by config.
src/sim/simulate.cc:199: info: Entering event queue @ 0.  Starting simulation...
src/cpu/kvm/perfevent.cc:217: panic: PerfKvmCounter::attach failed (2)
Memory Usage: 3793480 KBytes
Program aborted at tick 0
src/cpu/kvm/perfevent.cc:217: panic: PerfKvmCounter::attach failed (2)
Memory Usage: 3793488 KBytes
Aborted (core dumped)
```

The test config used is the following:
```
from gem5.coherence_protocol import CoherenceProtocol
from gem5.components.boards.x86_board import X86Board
from gem5.components.memory.single_channel import SingleChannelDDR3_1600
from gem5.components.processors.cpu_types import CPUTypes
from gem5.components.processors.simple_processor import (
    SimpleProcessor
)
from gem5.components.processors.simple_switchable_processor import (
    SimpleSwitchableProcessor,
)
from gem5.isas import ISA
from gem5.resources.resource import obtain_resource
from gem5.simulate.exit_event import ExitEvent
from gem5.simulate.simulator import Simulator
from gem5.utils.requires import requires

# This runs a check to ensure the gem5 binary is compiled to X86 and to the
# MESI Two Level coherence protocol.
requires(
    isa_required=ISA.X86,
    coherence_protocol_required=CoherenceProtocol.MESI_TWO_LEVEL,
)

from gem5.components.cachehierarchies.ruby.mesi_two_level_cache_hierarchy import (
    MESITwoLevelCacheHierarchy,
)

# Here we setup a MESI Two Level Cache Hierarchy.
cache_hierarchy = MESITwoLevelCacheHierarchy(
    l1d_size="16kB",
    l1d_assoc=8,
    l1i_size="16kB",
    l1i_assoc=8,
    l2_size="256kB",
    l2_assoc=16,
    num_l2_banks=1,
)

# Setup the system memory.
memory = SingleChannelDDR3_1600(size="3GB")

# Here we setup the processor. This is a special switchable processor in which
# a starting core type and a switch core type must be specified. Once a
# configuration is instantiated a user may call `processor.switch()` to switch
# from the starting core types to the switch core types. In this simulation
# we start with KVM cores to simulate the OS boot, then switch to the Timing
# cores for the command we wish to run after boot.
processor = SimpleProcessor(
    cpu_type=CPUTypes.KVM,
    isa=ISA.X86,
    num_cores=2,
)

# Here we setup the board. The X86Board allows for Full-System X86 simulations.
board = X86Board(
    clk_freq="3GHz",
    processor=processor,
    memory=memory,
    cache_hierarchy=cache_hierarchy,
)

# Here we set the Full System workload.
# The `set_kernel_disk_workload` function for the X86Board takes a kernel, a
# disk image, and, optionally, a command to run.


workload = obtain_resource("x86-ubuntu-22.04-boot-with-systemd")
# command = (
#     "m5 exit;"
# )
# workload.set_parameter("readfile_contents", command)

board.set_workload(workload)

def exit_event_handler():
    print("first exit event")
    yield False
    print("second exit event")
    yield False
    print("third exit event")
    yield False
    print("fourth exit event")
    yield False
    print("fifth exit event")
    yield True

simulator = Simulator(
    board=board,
    on_exit_event={
        # Here we want override the default behavior for the first m5 exit
        # exit event. Instead of exiting the simulator, we just want to
        # switch the processor. The 2nd m5 exit after will revert to using
        # default behavior where the simulator run will exit.
        ExitEvent.EXIT: exit_event_handler()
    },
)
simulator.run()
```

The above is an test script that should get 3 exits.
**Affects version**
develop for 24.0

**To Reproduce**
Steps to reproduce the behavior. Please assume starting from a clean repository:

1. Compile gem5 with command `scons build/X86/gem5.opt`
2. Run the test config above or any fs simulation the uses x86 kvm

**Expected behavior**
We should get 3 m5 exit events.

**Comment by powerjg** [comment]
Have you tried setting "allow_hybrid" to false on the KVM object?

**Comment by Harshil2107** [comment]
No, I have not tried that

**Comment by Harshil2107** [comment]
Changing https://github.com/gem5/gem5/blob/10b24dc9a477d9c3b82c1a4c86c7462d65d3a64d/src/cpu/kvm/BaseKvmCPU.py#L76

to false allows me to run simulation again.

**Comment by powerjg** [comment]
Related PRs: 
- #1127 (revert 1065)
- #1138 (some other fix?)
- #1065 (enable KVM on hybrid architectures)
- #1040 (some fix?)

The data we have right now is that on an i7-13700 (I think; definitely 13th gen with both P and E cores) after #1065 KVM breaks. KVM works on that machine with gem5 stable.

Given that #1065 and #1040 were merged around the same time, it's unclear which one (if it's either or both or one of them) that are causing the regression.

@Harshil2107, please reply to this with which commits work and don't work on the three machines we talked about (Intel with P/E cores, Intel without P/E cores, and AMD). Please give all the information you have now, and fill in more as you get it.

**Comment by Harshil2107** [comment]
I will keep updating this message as I get updates

| Machine| Stable | Devlop | Cherry picked 1065 on stable | Cherry picked 1040 on stable | Cherry picked  1065 + 1040 on stabe |
|----------|----------|----------|----------|----------|----------|
| Intel with P/E cores | KVM works | KVM fails with error 1 | KVM fails with error 1 | KVM fails with error 2 |KVM fails with error 1 |
| AMD | KVM works | KVM failes with error 2 | KVM works |KVM fails with error 2 | KVM fails with error 2 |
| Intel without P/E cores | KVM works | KVM failes with error 2 |  KVM works |KVM fails with error 2  |KVM fails with error 2 |


KVM has failed with 2 different errors so far I will refer to them as error 1 and error 2

Error 1: 
```

info: Using default config
warn: Resource x86-ubuntu-22.04-img with version 1.0.0 is not known to be compatible with gem5 version 23.1.0.0. This may cause problems with your simulation. This resource's compatibility with different gem5 versions can be found here: https://resources.gem5.org/resources/x86-ubuntu-22.04-img/versions
warn: Resource x86-ubuntu-22.04-img with version 1.0.0 is not known to be compatible with gem5 version 23.1.0.0. This may cause problems with your simulation. This resource's compatibility with different gem5 versions can be found here: https://resources.gem5.org/resources/x86-ubuntu-22.04-img/versions
Resource 'x86-ubuntu-22.04-img' was not found locally. Downloading to '/home/harshilp/.cache/gem5/x86-ubuntu-22.04-img.gz'...
Finished downloading resource 'x86-ubuntu-22.04-img'.
Decompressing resource 'x86-ubuntu-22.04-img' ('/home/harshilp/.cache/gem5/x86-ubuntu-22.04-img.gz')...
Finished decompressing resource 'x86-ubuntu-22.04-img'.
warn: <orphan X86Board>.kvm_vm already has parent not resetting parent.
        Note: kvm_vm is not a parameter of X86Board
warn: (Previously declared as <orphan X86Board>.processor.kvm_vm
Global frequency set at 1000000000000 ticks per second
warn: No dot file generated. Please install pydot to generate the dot file and pdf.
src/mem/dram_interface.cc:690: warn: DRAM device capacity (8192 Mbytes) does not match the address range assigned (4096 Mbytes)
src/sim/kernel_workload.cc:46: info: kernel located at: /home/harshilp/.cache/gem5/x86-linux-kernel-5.4.0-105-generic
src/base/statistics.hh:279: warn: One of the stats is a legacy stat. Legacy stat is a stat that does not belong to any statistics::Group. Legacy stat is deprecated.
      0: board.pc.south_bridge.cmos.rtc: Real-time clock set to Sun Jan  1 00:00:00 2012
board.pc.com_1.device: Listening for connections on port 3456
src/base/statistics.hh:279: warn: One of the stats is a legacy stat. Legacy stat is a stat that does not belong to any statistics::Group. Legacy stat is deprecated.
src/dev/intel_8254_timer.cc:128: warn: Reading current count from inactive timer.
board.remote_gdb: Listening for connections on port 7000
src/cpu/kvm/base.cc:170: info: KVM: Coalesced MMIO disabled by config.
src/cpu/kvm/base.cc:170: info: KVM: Coalesced MMIO disabled by config.
src/sim/simulate.cc:199: info: Entering event queue @ 0.  Starting simulation...
src/cpu/kvm/perfevent.cc:217: panic: PerfKvmCounter::attach failed (2)
Memory Usage: 3791968 KBytes
src/cpu/kvm/perfevent.cc:217: panic: PerfKvmCounter::attach failed (2)
Memory Usage: 3791968 KBytes
Program aborted at tick 0
Aborted (core dumped)
```
Error 2
```
info: Using default config
warn: Resource x86-ubuntu-22.04-img with version 1.0.0 is not known to be compatible with gem5 version 23.1.0.0. This may cause problems with your simulation. This resource's compatibility with different gem5 versions can be found here: https://resources.gem5.org/resources/x86-ubuntu-22.04-img/versions
warn: Resource x86-ubuntu-22.04-img with version 1.0.0 is not known to be compatible with gem5 version 23.1.0.0. This may cause problems with your simulation. This resource's compatibility with different gem5 versions can be found here: https://resources.gem5.org/resources/x86-ubuntu-22.04-img/versions
Resource 'x86-ubuntu-22.04-img' was not found locally. Downloading to '/home/harshilp/.cache/gem5/x86-ubuntu-22.04-img.gz'...
Finished downloading resource 'x86-ubuntu-22.04-img'.
Decompressing resource 'x86-ubuntu-22.04-img' ('/home/harshilp/.cache/gem5/x86-ubuntu-22.04-img.gz')...
Finished decompressing resource 'x86-ubuntu-22.04-img'.
warn: <orphan X86Board>.kvm_vm already has parent not resetting parent.
	Note: kvm_vm is not a parameter of X86Board
warn: (Previously declared as <orphan X86Board>.processor.kvm_vm
Global frequency set at 1000000000000 ticks per second
src/mem/dram_interface.cc:690: warn: DRAM device capacity (8192 Mbytes) does not match the address range assigned (4096 Mbytes)
src/sim/kernel_workload.cc:46: info: kernel located at: /home/harshilp/.cache/gem5/x86-linux-kernel-5.4.0-105-generic
src/base/statistics.hh:279: warn: One of the stats is a legacy stat. Legacy stat is a stat that does not belong to any statistics::Group. Legacy stat is deprecated.
      0: board.pc.south_bridge.cmos.rtc: Real-time clock set to Sun Jan  1 00:00:00 2012
board.pc.com_1.device: Listening for connections on port 3457
src/base/statistics.hh:279: warn: One of the stats is a legacy stat. Legacy stat is a stat that does not belong to any statistics::Group. Legacy stat is deprecated.
src/dev/intel_8254_timer.cc:128: warn: Reading current count from inactive timer.
board.remote_gdb: Listening for connections on port 7001
src/cpu/kvm/base.cc:169: info: KVM: Coalesced MMIO disabled by config.
src/cpu/kvm/base.cc:169: info: KVM: Coalesced MMIO disabled by config.
src/sim/simulate.cc:199: info: Entering event queue @ 0.  Starting simulation...
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010200) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010202) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010204) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010206) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010208) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc001020a) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010201) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010203) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010205) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010207) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010209) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc001020b) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x12) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x11) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x4b564d01) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x4b564d00) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000000) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000001) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000020) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000021) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000022) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000023) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000100) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000101) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000102) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000103) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000104) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000105) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000003) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000002) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000010) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000080) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x400000b0) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000073) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000106) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000107) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000108) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x4b564d02) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x4b564d03) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x4b564d04) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x3b) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x6e0) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x10a) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x1a0) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x9e) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x34) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xce) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x140) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc001011f) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x1fc) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010015) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x4b564d05) unsupported by gem5. Skipping.
src/dev/x86/pc.cc:117: warn: Don't know what interrupt to clear for console.
src/arch/x86/kvm/x86_cpu.cc:563: warn: cs limit (0xffffffff) and g (0) combination is illegal.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010200) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010202) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010204) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010206) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010208) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc001020a) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010201) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010203) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010205) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010207) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010209) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc001020b) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x12) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x11) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x4b564d01) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x4b564d00) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000000) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000001) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000020) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000021) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000022) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000023) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000100) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000101) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000102) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000103) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000104) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000105) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000003) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000002) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000010) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000080) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x400000b0) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000073) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000106) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000107) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x40000108) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x4b564d02) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x4b564d03) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x4b564d04) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x3b) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x6e0) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x10a) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x1a0) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x9e) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x34) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xce) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x140) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc001011f) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x1fc) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0xc0010015) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1683: warn: kvm-x86: MSR (0x4b564d05) unsupported by gem5. Skipping.
src/arch/x86/kvm/x86_cpu.cc:1725: panic: KVM: Failed to set guest debug registers
Memory Usage: 3793860 KBytes
Program aborted at tick 608000000058
--- BEGIN LIBC BACKTRACE ---
build/X86/gem5.opt(_ZN4gem515print_backtraceEv+0x30)[0x56340fb9c600]
build/X86/gem5.opt(_ZN4gem512abortHandlerEi+0x4e)[0x56340fbc1fae]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x14420)[0x7f4aa828c420]
/lib/x86_64-linux-gnu/libc.so.6(gsignal+0xcb)[0x7f4aa745600b]
/lib/x86_64-linux-gnu/libc.so.6(abort+0x12b)[0x7f4aa7435859]
build/X86/gem5.opt(+0x8f8e15)[0x56340f0d3e15]
build/X86/gem5.opt(+0x124c7fd)[0x56340fa277fd]
build/X86/gem5.opt(_ZN4gem59X86KvmCPU18updateKvmStateXCRsEv+0xe8)[0x56340fa27918]
build/X86/gem5.opt(_ZN4gem59X86KvmCPU14updateKvmStateEv+0x50)[0x56340fa2c640]
build/X86/gem5.opt(_ZN4gem510BaseKvmCPU4tickEv+0x160)[0x56340f323bd0]
build/X86/gem5.opt(_ZN4gem510EventQueue10serviceOneEv+0xc2)[0x56340fbb0fa2]
build/X86/gem5.opt(_ZN4gem59doSimLoopEPNS_10EventQueueE+0x70)[0x56340fbdd020]
build/X86/gem5.opt(_ZNSt6thread11_State_implINS_8_InvokerISt5tupleIJZN4gem516SimulatorThreads17runUntilLocalExitEvEUlPNS3_10EventQueueEE_S6_EEEEE6_M_runEv+0xbb)[0x56340fbde3bb]
/lib/x86_64-linux-gnu/libstdc++.so.6(+0xd6df4)[0x7f4aa7847df4]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x8609)[0x7f4aa8280609]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x43)[0x7f4aa7532353]
--- END LIBC BACKTRACE ---
For more info on how to address this issue, please visit https://www.gem5.org/documentation/general_docs/common-errors/ 

Aborted (core dumped)
```

**Comment by abmerop** [comment]
#1138 fixes "Error 2".  Not sure about the other error, have not seen it.

**Comment by powerjg** [comment]
@Harshil2107 can you run with Matt's new PR (#1138) and see what happens?

If all cases fail with #1065, I think we should revert it until @nmosier can resubmit.

**Comment by Harshil2107** [comment]
I will run #1138 on the AMD machine as KVM works with #1065 on that macine

**Comment by Harshil2107** [comment]
The script with KVM runs on AMD when I checkedout #1138
